test/S3Plugin.test.js

```diff
@@ -0,0 +1,244 @@
+const
+  KuzzleErrors = require('kuzzle-common-objects').errors,
+  mockrequire = require('mock-require'),
+  sinon = require('sinon'),
+  { 
+    BadRequestError,
+    NotFoundError,
+    InternalError: KuzzleInternalError 
+  } = require('kuzzle-common-objects').errors,
+  should = require('should');
+
+class S3Mock {
+  constructor (config) {
+    this.config = config;
+  }
+}
+
+function s3Resolve (resolveArg) {
```

Comment:
It's just syntactic sugar, otherwise I would have to write `sinon.stub().returns({ promise: () => Promise.resolve('url') })` because the S3 SDK use a very weird way to return promises. (eg `someFunction().promise()`